### PR TITLE
SNOW-683054: Analyzer-related test coverage improvement

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/_internal/analyzer/analyzer.py [hunk 1]
```diff
@@ -598,9 +584,6 @@ def do_resolve_with_resolved_children(
             )
 
         if isinstance(logical_plan, Pivot):
-            if len(logical_plan.aggregates) != 1:
-                raise ValueError("Only one aggregate is supported with pivot")
-
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/_internal/analyzer/analyzer.py:603]: Why not add a negative test instead of removing this check?

Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/_internal/analyzer/analyzer.py:603]: forgot this one - already checked here https://github.com/snowflakedb/snowpark-python/blob/e3117c91a91c678d41ea77824bdd55272429b388/src/snowflake/snowpark/relational_grouped_dataframe.py#L177

File: src/snowflake/snowpark/_internal/analyzer/unary_plan_node.py [hunk 2]
```diff
@@ -23,22 +23,14 @@ def __init__(
...
 
 
 class Sort(UnaryNode):
-    def __init__(
-        self, order: List[SortOrder], is_global: bool, child: LogicalPlan
-    ) -> None:
+    def __init__(self, order: List[SortOrder], child: LogicalPlan) -> None:
         super().__init__(child)
         self.order = order
-        self.is_global = is_global
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/_internal/analyzer/unary_plan_node.py:41]: I believe `is_global` is never used?

Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/_internal/analyzer/unary_plan_node.py:41]: yea it's not used

**Review by sfc-gh-aalam — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/_internal/analyzer/analyzer.py [hunk 3]
```diff
@@ -579,16 +572,9 @@ def do_resolve_with_resolved_children(
...
-                on_top_of_order_by = True
-            elif (
-                isinstance(logical_plan.child, SnowflakePlan)
-                and logical_plan.child.source_plan
-            ):
-                on_top_of_order_by = isinstance(logical_plan.child.source_plan, Sort)
-            else:
-                on_top_of_order_by = False
-
+            on_top_of_order_by = isinstance(
```
[the remark this answers is not in the document]
Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/_internal/analyzer/analyzer.py:575]: It's by design - we have it in type hints https://github.com/snowflakedb/snowpark-python/blob/f38f1f5dd7e7c1cb8e55fc7233907c5a2dab642d/src/snowflake/snowpark/_internal/analyzer/unary_plan_node.py#L12. And I verified that how we construct Sort() https://github.com/snowflakedb/snowpark-python/blob/f38f1f5dd7e7c1cb8e55fc7233907c5a2dab642d/src/snowflake/snowpark/dataframe.py#L1192

File: src/snowflake/snowpark/_internal/analyzer/select_statement.py [hunk 4]
```diff
@@ -756,7 +758,8 @@ def can_projection_dependent_columns_be_flattened(
     dependent_columns: Optional[Set[str]], subquery_column_states: ColumnStateDict
 ) -> bool:
     can_be_flattened = True
-    if dependent_columns == COLUMN_DEPENDENCY_DOLLAR:
+    # COLUMN_DEPENDENCY_DOLLAR should already be handled before calling this function
```
[the remark this answers is not in the document]
Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/_internal/analyzer/select_statement.py:761]: I'm not sure about the original intention for this code. @sfc-gh-yixie

**Review by sfc-gh-yixie — APPROVED** [review]

**Pull request merged**